core/node/libp2p/rcmgr.go

```diff
@@ -133,540 +117,245 @@ func ResourceManager(cfg config.SwarmConfig) interface{} {
+		case limitNameStreams:
+			ri.LimitName = limitNameStreams
+			ri.Limit = int64(limit.Streams)
+			ri.CurrentUsage = int64(stat.Streams)
+		case limitNameStreamsInbound:
+			ri.LimitName = limitNameStreamsInbound
+			ri.Limit = int64(limit.StreamsInbound)
+			ri.CurrentUsage = int64(stat.StreamsInbound)
+		case limitNameStreamsOutbound:
+			ri.LimitName = limitNameStreamsOutbound
+			ri.Limit = int64(limit.StreamsOutbound)
+			ri.CurrentUsage = int64(stat.StreamsOutbound)
+		case limitNameConns:
+			ri.LimitName = limitNameConns
+			ri.Limit = int64(limit.Conns)
+			ri.CurrentUsage = int64(stat.Conns)
+		case limitNameConnsInbound:
+			ri.LimitName = limitNameConnsInbound
+			ri.Limit = int64(limit.ConnsInbound)
+			ri.CurrentUsage = int64(stat.ConnsInbound)
+		case limitNameConnsOutbound:
+			ri.LimitName = limitNameConnsOutbound
+			ri.Limit = int64(limit.ConnsOutbound)
+			ri.CurrentUsage = int64(stat.ConnsOutbound)
+		case limitNameFD:
+			ri.LimitName = limitNameFD
+			ri.Limit = int64(limit.FD)
+			ri.CurrentUsage = int64(stat.FD)
+		case limitNameMemory:
+			ri.LimitName = limitNameMemory
+			ri.Limit = int64(limit.Memory)
+			ri.CurrentUsage = int64(stat.Memory)
 		}
 
-		err = mgr.ViewPeer(pid, func(s network.PeerScope) error { return setLimit(s, defaults.PeerDefault.Build(rcmgr.BaseLimit{})) })
-		setConfigFunc = func() rcmgr.BaseLimit {
-			if configLimits.Peer == nil {
-				configLimits.Peer = map[peer.ID]rcmgr.ResourceLimits{}
-			}
-			configLimits.Peer[pid] = defaults.PeerDefault
-
-			return defaults.PeerDefault.Build(rcmgr.BaseLimit{})
+		if ri.Limit == 0 {
+			continue
 		}
-	default:
-		return result, fmt.Errorf("invalid scope %q", scope)
-	}
 
-	if err != nil {
-		return result, fmt.Errorf("resetting new limits on resource manager: %w", err)
+		result = append(result, ri)
 	}
 
-	result = setConfigFunc()
+	return result
+}
 
-	if err := repo.SetConfig(cfg); err != nil {
-		return result, fmt.Errorf("writing new limits to repo config: %w", err)
+func scopeStatToBaseLimit(ss network.ScopeStat) rcmgr.BaseLimit {
+	return rcmgr.BaseLimit{
+		Streams:         ss.NumStreamsInbound + ss.NumStreamsOutbound,
+		StreamsInbound:  ss.NumStreamsInbound,
+		StreamsOutbound: ss.NumStreamsOutbound,
+		Conns:           ss.NumConnsInbound + ss.NumConnsOutbound,
+		ConnsInbound:    ss.NumConnsInbound,
+		ConnsOutbound:   ss.NumConnsOutbound,
+		FD:              ss.NumFD,
+		Memory:          ss.Memory,
 	}
-
-	return result, nil
 }
 
-func ensureConnMgrMakeSenseVsResourceMgr(orig rcmgr.ConcreteLimitConfig, cmgr config.ConnMgr) error {
-	if cmgr.Type.WithDefault(config.DefaultConnMgrType) == "none" {
-		return nil // none connmgr, no checks to do
+func ensureConnMgrMakeSenseVsResourceMgr(concreteLimits rcmgr.ConcreteLimitConfig, cfg config.SwarmConfig) error {
+	if cfg.ConnMgr.Type.WithDefault(config.DefaultConnMgrType) == "none" || len(cfg.ResourceMgr.Allowlist) != 0 {
+		return nil // none connmgr, or setup with an allow list, no checks to do
```

Comment:
@BigLep you could have a config where you have a large lists of trusted node, lists that is larger than your connection manager options.
While you have defaults limits set to `blockAll`.

Reply:
ACK.  Makes sense.  I think it's worth adding a comment here.  Maybe something like...

```
If an allowlist is set, a user may be enacting some form of DoS defense.
We don't want want to modify the System.ConnsInbound in that case as it may make sense for it to be (and stay) as "blockAll" so that only connections with the allow list multiaddr get established.  
```